### ISSUE TITLE
rsync 3.2.1-1 crashes on Debian bullseye/sid (testing)

### ISSUE THREAD

**Issue by filipsPL** [WayneD/rsync#43]
Hi, first of all, thanks for maintaining the rsync software!

Bug: since upgrade from 3.1.3-6 to 3.2.1-1 on the server (Debian bullseye/sid (testing)) I'm not able to rsync.

On the client side:

```bash
$rsync -av --rsync-path=/home/user/rsync-debug --password-file=pa.txt file.png  rsync://user@domain/path
rsync: didn't get server startup line
rsync error: error starting client-server protocol (code 5) at main.c(1675) [sender=3.1.3]
```

Server syslog:
```
Jun 29 09:06:49 localhost kernel: [2027341.018850] rsync[3763422]: segfault at 559cc8cefe4f ip 0000559cc8cda451 sp 00007fff4a01e830 error 7 in rsync[559cc8c96000+4e000]
Jun 29 09:06:49 localhost kernel: [2027341.018874] Code: 01 80 fb 25 75 d3 80 7d 01 25 0f 84 a9 00 00 00 be 25 00 00 00 4c 89 f7 4c 89 04 24 e8 18 c4 fb ff 4c 8b 04 24 48 85 c0 74 af <c6> 00 00 4c 89 f7 4c 89 44 24 08 48 89 04 24 e8 5b c
4 fb ff 48 8b
```

When downgrading to 3.1.3-6 everything works fine.

**Comment by WayneD** [comment]
What CPU architecture is that?  Could you enable core dumps and use gdb on the dump to get a backtrace? I've used the latest rsync on Debian bullseye/sid and it works fine.

**Comment by filipsPL** [comment]
Please find the details below. I didn't use gdb as I can't make it work.

```bash
# cat /proc/cpuinfo | grep "model name"
model name      : Intel(R) Atom(TM) CPU  330   @ 1.60GHz

# uname -a
Linux kofola2 5.6.0-2-amd64 #1 SMP Debian 5.6.14-1 (2020-05-23) x86_64 GNU/Linux


# rsync --version
rsync  version 3.2.1  protocol version 31
Copyright (C) 1996-2020 by Andrew Tridgell, Wayne Davison, and others.
Web site: https://rsync.samba.org/
Capabilities:
    64-bit files, 64-bit inums, 64-bit timestamps, 64-bit long ints,
    socketpairs, hardlinks, symlinks, IPv6, atimes, batchfiles, inplace,
    append, ACLs, xattrs, optional protect-args, iconv, symtimes, prealloc
Checksum list:
    xxh64 (xxhash) md5 md4 none
Compress list:
    zstd lz4 zlibx zlib none

rsync comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
are welcome to redistribute it under certain conditions.  See the GNU
General Public Licence for details.


# ldd /usr/bin/rsync
        linux-vdso.so.1 (0x00007ffd558df000)
        libacl.so.1 => /lib/x86_64-linux-gnu/libacl.so.1 (0x00007fbcae67f000)
        libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fbcae662000)
        libpopt.so.0 => /lib/x86_64-linux-gnu/libpopt.so.0 (0x00007fbcae654000)
        liblz4.so.1 => /lib/x86_64-linux-gnu/liblz4.so.1 (0x00007fbcae632000)
        libzstd.so.1 => /lib/x86_64-linux-gnu/libzstd.so.1 (0x00007fbcae555000)
        libxxhash.so.0 => /lib/x86_64-linux-gnu/libxxhash.so.0 (0x00007fbcae54b000)
        libcrypto.so.1.1 => /lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007fbcae25d000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fbcae09a000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fbcae095000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fbcae074000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fbcae720000)



ulimit -c unlimited
strace -f -t -s 1024 -o /tmp/rsync-$$.out rsync --daemon --no-detach
```

[rsync-3141621.out](https://github.com/WayneD/rsync/files/4849902/rsync-3141621.out.txt)

**Comment by WayneD** [comment]
Much appreciated! I'm giving that a look now.  Please note that you should change your daemon password, as the strace showed the contents of your secrets file.

**Comment by WayneD** [comment]
I've figured out the crash and am including a fix in 3.2.2.  In the meantime, you can avoid the crash if you tweak your daemon config file and explicitly define the `log format` value instead of letting it default:

`log format = %o %h [%a] %m (%u) %f %l`
